src/detector/implementations/picamera2_implementation.py

```diff
@@ -0,0 +1,181 @@
+
+        if self._captured_image is None:
+            message: str = "Failed to grab frame."
+            self._status.capture_errors.append(message)
+            self._capture_status.status = CaptureStatus.Status.FAILURE
+            return ("error", message)
+
+        self._captured_timestamp_utc = datetime.datetime.utcnow()
+
+    def set_capture_device(self, **kwargs) -> EmptyResponse | ErrorResponse:
+        
+        self._capture = self._detect_os_and_open_video()
+        if self._capture is None:
+            return ErrorResponse(message=f"Failed to open capture device")
+        
+        default_brightness = self._camera.camera_controls['Brightness'][2]
+        default_contrast = self._camera.camera_controls['Contrast'][2]
+        default_sharpness = self._camera.camera_controls['Sharpness'][2]
+        default_auto_exposure = self._camera.camera_controls['AeEnable'][2]
+        default_exposure = self._camera.camera_controls['ExposureValue'][2]
+
+        self._camera.controls.Brightness = default_brightness
+        self._camera.controls.Contrast = default_contrast
+        self._camera.controls.Sharpness = default_sharpness
+        self._camera.controls.AeEnable = default_auto_exposure
+        self._camera.controls.ExposureValue = default_exposure
+        
+        return EmptyResponse()
+
+    # noinspection DuplicatedCode
+    def set_capture_properties(self, **kwargs) -> EmptyResponse:
+        """
+        :key request: SetCapturePropertiesRequest
+        """
+
+        request: SetCapturePropertiesRequest = get_kwarg(
+            kwargs=kwargs,
+            key="request",
+            arg_type=SetCapturePropertiesRequest)
+
+        if self._capture is not None:
+
+            # FPS and resolution change require the camera be off
+            # While other settings require the camera be on
+            self._camera.stop()
+            if request.resolution_x_px is not None and request.resolution_y_px is not None:
+                self._camera.video_configuration.size = (request.resolution_x_px,request.resolution_y_px)
+            if request.fps is not None:
+                self._camera.video_configuration.controls.FrameRate = request.fps
+            self._camera.configure("video")
+            self._camera.start()
+
+            if request.auto_exposure is not None:
+                self._camera.controls.AeEnable = request.auto_exposure
+            # TODO: how to enforce values be entered in the proper range?
+            if request.exposure is not None:
+                self._camera.controls.ExposureValue = request.exposure
+            if request.brightness is not None:
+                self._camera.controls.Brightness = request.brightness
+            if request.contrast is not None:
+                self._camera.controls.Contrast = request.contrast
+            if request.sharpness is not None:
+                self._camera.controls.Sharpness = request.sharpness
```

Comment:
Are there other settings beyond these ones? I'm not asking for changes, just information

Reply:
Yes there are, a full list can be found [here](https://datasheets.raspberrypi.com/camera/picamera2-manual.pdf#camera-controls)

Reply:
Thanks. There are certainly a number of interesting settings there. It looks to me like the exposure used in OpenCV and the exposure used in picamera possibly use different conventions. I think we can leave this as-is for now, but we will for sure later want to have different input data structures depending on whether the OpenCV or PiCamera driver is being used.

Reply:
Defer this to later:  #32